encrypted-media/scripts/playback-temporary-waitingforkey.js

```diff
@@ -0,0 +1,90 @@
+function runTest(config) {
+
+    var testname = config.keysystem + ', successful playback, temporary, '
+                                    + /video\/([^;]*)/.exec( config.videoType )[ 1 ]
+                                    + ', waitingforkey event';
+
+    var configuration = {   initDataTypes: [ config.initDataType ],
+                            audioCapabilities: [ { contentType: config.audioType } ],
+                            videoCapabilities: [ { contentType: config.videoType } ],
+                            sessionTypes: [ 'temporary' ] };
+
+    async_test( function( test ) {
+
+        var _video = config.video,
+            _mediaKeys,
+            _mediaKeySessions = [ ],
+            _mediaSource;
+
+        function onMessage(event) {
+            assert_any( assert_equals, event.target, _mediaKeySessions );
+            assert_true( event instanceof window.MediaKeyMessageEvent );
+            assert_equals( event.type, 'message');
+
+            assert_any( assert_equals,
+                        event.messageType,
+                        [ 'license-request', 'individualization-request' ] );
+
+            config.messagehandler( config.keysystem, event.messageType, event.message )
+            .then( function( response ) {
+
+                event.target.update( response )
+                .catch(function(error) {
+                    forceTestFailureFromPromise(test, error);
+                });
+            });
+        }
+
+        function onWaitingForKey(event) {
+
+            // Not using waitForEventAndRunStep() to avoid too many
+            // EVENT(onTimeUpdate) logs.
+            _video.addEventListener('timeupdate', onTimeupdate, true);
+
+            // Create sessions now if we do not have them
+            if ( _mediaKeySessions.length === 0 ) {
+                config.initData.forEach( function( initData ) {
+
+                    var mediaKeySession = _mediaKeys.createSession( 'temporary' );
```

Comment:
Spaces inside () are applied inconsistently. For example, here vs. line 42. I prefer no spaces for simplicity and slightly shorter lines, but let's pick one and be consistent.